shared/rpcwire/websocket.go

```
@@ -0,0 +1,278 @@
+
+func NewWebSocketTransport() WebSocketTransport {
+	return WebSocketTransport{}
+}
+
+func (WebSocketTransport) Dial(ctx context.Context, endpoint Endpoint) (Conn, error) {
+	rawConn, err := dialWebSocketEndpoint(ctx, endpoint)
+	if err != nil {
+		return nil, err
+	}
+	adapter := newWebSocketConn()
+	socket, _, err := dialWebSocketClientContext(ctx, rawConn, endpoint, adapter)
```

golangci-lint (2.11.4)

[error] response body must be closed

(bodyclose)